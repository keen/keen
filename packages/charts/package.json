{
  "name": "@keen.io/charts",
  "version": "2.7.12",
  "description": "Keen charts",
  "author": "Keen.IO <team@keen.io> (https://keen.io/)",
  "homepage": "https://github.com/keen/keen#readme",
  "license": "MIT",
  "main": "dist/index.js",
  "types": "dist/index.d.ts",
  "module": "src/index.ts",
  "files": [
    "dist"
  ],
  "repository": {
    "type": "git",
    "url": "git+https://github.com/keen/keen.git"
  },
  "scripts": {
    "test": "jest",
    "prebuild": "rimraf dist && rimraf tsconfig.tsbuildinfo",
    "build": "tsc"
  },
  "bugs": {
    "url": "https://github.com/keen/keen/issues"
  },
  "peerDependencies": {
    "react": ">=16",
    "react-dom": ">=16",
    "styled-components": ">=5"
  },
  "dependencies": {
    "@keen.io/charts-utils": "^1.5.0",
    "@keen.io/colors": "^1.1.0",
    "@keen.io/react-hooks": "^1.0.7",
    "@keen.io/ui-core": "^2.14.2",
    "column-resizer": "^1.3.5",
    "d3-array": "^2.4.0",
    "d3-axis": "^1.0.12",
    "d3-drag": "^1.2.5",
    "d3-fetch": "^1.1.2",
    "d3-geo": "^1.11.9",
    "d3-interpolate": "^1.4.0",
    "d3-polygon": "^1.0.6",
    "d3-scale": "^3.2.1",
    "d3-selection": "^1.4.1",
    "d3-shape": "^1.3.7",
    "d3-time-format": "^2.2.3",
    "d3-transition": "^1.3.2",
    "d3-zoom": "^1.8.3",
    "framer-motion": "^1.11.1",
    "numeral": "^2.0.6",
    "polished": "^3.4.4",
    "react-measure": "^2.3.0",
    "use-debounce": "^3.3.0",
    "uuid": "^8.3.2"
  },
  "devDependencies": {
    "@keen.io/storybook-utils": "^2.0.58",
    "@types/d3-array": "^2.0.0",
    "@types/d3-axis": "^1.0.12",
    "@types/d3-drag": "^1.2.3",
    "@types/d3-fetch": "^1.1.5",
    "@types/d3-geo": "^1.11.1",
    "@types/d3-interpolate": "^1.3.1",
    "@types/d3-polygon": "^1.0.7",
    "@types/d3-scale": "^2.1.1",
    "@types/d3-selection": "^1.4.1",
    "@types/d3-shape": "^1.3.2",
    "@types/d3-time-format": "^2.1.1",
    "@types/d3-transition": "^1.1.6",
    "@types/d3-zoom": "^1.7.4",
    "@types/numeral": "^2.0.0",
    "@types/react-measure": "^2.0.5",
    "@types/styled-components": "^5.1.0",
    "@types/uuid": "^8.3.0"
  },
  "publishConfig": {
    "access": "public"
  }
}
