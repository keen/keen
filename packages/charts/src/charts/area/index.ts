import AreaChart, { Props as AreaChartSettings } from './area-chart.component';

export { AreaChart };
export type { AreaChartSettings };
