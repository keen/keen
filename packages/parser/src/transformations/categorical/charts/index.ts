import { categoricalChartTransformation } from './chart-transformation';
import { choroplethChartTransformation } from './choropleth-transformation';

export { categoricalChartTransformation, choroplethChartTransformation };
