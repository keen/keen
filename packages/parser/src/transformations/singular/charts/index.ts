import { tableChartTransformation } from './table-transformation';

export { tableChartTransformation };
