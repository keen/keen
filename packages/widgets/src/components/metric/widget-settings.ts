import { colors } from '@keen.io/colors';

import { widgetSettings } from '../../widget-settings';

export const metricWidgetSettings = {
  card: {
    backgroundColor: colors.white['500'],
    borderRadius: '0px',
    border: 'none',
    hasShadow: true,
  },
  title: widgetSettings.title,
  subtitle: widgetSettings.subtitle,
};
