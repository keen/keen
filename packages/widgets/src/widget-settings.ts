import { colors } from '@keen.io/colors';
import { LegendSettings, WidgetSettings } from './types';

export const legendSettings: LegendSettings = {
  enabled: true,
  position: 'top',
  alignment: 'left',
  layout: 'horizontal',
  typography: {
    fontSize: 10,
    fontFamily: 'Gangster Grotesk, sans-serif',
    fontStyle: 'normal',
    fontWeight: 'normal',
    fontColor: colors.black['500'],
  },
  card: undefined,
  title: undefined,
};

export const widgetSettings: WidgetSettings = {
  legend: legendSettings,
  card: {
    backgroundColor: colors.white['500'],
    borderRadius: 0,
    hasShadow: true,
    enabled: true,
  },
  title: {
    content: '',
    typography: {
      fontSize: 20,
      fontFamily: 'Gangster Grotesk, sans-serif',
      fontStyle: 'normal',
      fontWeight: 'normal',
      fontColor: colors.blue['500'],
    },
  },
  subtitle: {
    content: '',
    typography: {
      fontSize: 14,
      fontFamily: 'Lato, sans-serif',
      fontStyle: 'normal',
      fontWeight: 'normal',
      fontColor: colors.black['500'],
    },
  },
};
