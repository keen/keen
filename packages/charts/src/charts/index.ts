import { BarChart, BarChartSettings } from './bar';
import { LineChart, LineChartSettings } from './line';
import { AreaChart, AreaChartSettings } from './area';
import { PieChart, PieChartSettings } from './pie';
import { DonutChart, DonutChartSettings } from './donut';
import { MetricChart, MetricChartSettings } from './metric';
import { FunnelChart, FunnelChartSettings } from './funnel';
import { HeatmapChart, HeatmapChartSettings } from './heatmap';
import { BubbleChart, BubbleChartSettings } from './bubble';
import {
  ChoroplethChart,
  ChoroplethChartSettings,
  fetchMapTopology,
} from './choropleth';

export {
  LineChart,
  LineChartSettings,
  AreaChart,
  AreaChartSettings,
  BarChart,
  BarChartSettings,
  BubbleChart,
  BubbleChartSettings,
  ChoroplethChart,
  ChoroplethChartSettings,
  PieChart,
  PieChartSettings,
  DonutChart,
  DonutChartSettings,
  MetricChart,
  MetricChartSettings,
  FunnelChart,
  FunnelChartSettings,
  HeatmapChart,
  HeatmapChartSettings,
  fetchMapTopology,
};
