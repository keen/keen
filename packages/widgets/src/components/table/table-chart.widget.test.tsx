/* eslint-disable @typescript-eslint/no-empty-function */
import React from 'react';
import { render as rtlRender } from '@testing-library/react';
import { mockAllIsIntersecting } from 'react-intersection-observer/test-utils';
import { ChartEvents } from '@keen.io/charts';
import 'jest-styled-components';

jest.mock('@keen.io/charts', () => ({
  ...jest.requireActual('@keen.io/charts'),
  ChartEvents: jest.fn().mockImplementation(() => ({
    subscribe: jest.fn().mockImplementation(() => function () {}),
  })),
}));

import { TableChartWidget } from './table-chart.widget';

import { widgetSettings } from '../../widget-settings';
import { chartData } from './table-chart.widget.fixtures';

const render = (overProps: any = {}) => {
  const props = {
    data: chartData,
    ...widgetSettings,
    ...overProps,
  };

  const wrapper = rtlRender(<TableChartWidget {...props} />);

  return {
    wrapper,
    props,
  };
};

mockAllIsIntersecting(true);

beforeEach(() => jest.clearAllMocks());

test('do not renders widget heading section', () => {
  const {
    wrapper: { queryByTestId },
  } = render();

  expect(queryByTestId('widget-heading')).not.toBeInTheDocument();
});

test('renders widget title', () => {
  const title = {
    ...widgetSettings.title,
    content: 'title',
  };
  const {
    wrapper: { getByText },
  } = render({ title });
  const titleElement = getByText(title.content);

  expect(titleElement).toBeInTheDocument();
  expect(titleElement).toMatchSnapshot();
});

test('renders widget subtitle', () => {
  const subtitle = {
    ...widgetSettings.subtitle,
    content: 'subtitle',
  };
  const {
    wrapper: { getByText },
  } = render({ subtitle });
  const titleElement = getByText(subtitle.content);

  expect(titleElement).toBeInTheDocument();
  expect(titleElement).toMatchSnapshot();
});

test('renders table plot component', () => {
  const {
    wrapper: { getByTestId },
  } = render();

  expect(getByTestId('table-chart-plot')).toBeInTheDocument();
});

test('creates ChartEvents class instance when table is rendered in edit mode', () => {
  const eventBus = jest.fn();

  const {} = render({
    inEditMode: true,
    eventBus,
  });

  expect(ChartEvents).toHaveBeenCalledTimes(1);
});

test('do not creates ChartEvents class instance', () => {
  const eventBus = jest.fn();

  const {} = render({
    inEditMode: false,
    eventBus,
  });

  expect(ChartEvents).not.toHaveBeenCalled();
});
