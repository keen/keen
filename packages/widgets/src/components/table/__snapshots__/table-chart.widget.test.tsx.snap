// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders widget subtitle 1`] = `
.c0 {
  font-size: 14px;
  font-family: Lato,sans-serif;
  font-style: normal;
  font-weight: normal;
  color: #1D2729;
  margin: 0;
  padding: 0;
}

<div
  class="c0"
  font-family="Lato, sans-serif"
  font-size="14"
  font-style="normal"
  font-weight="normal"
>
  subtitle
</div>
`;

exports[`renders widget title 1`] = `
.c0 {
  font-size: 20px;
  font-family: Gangster Grotesk,sans-serif;
  font-style: normal;
  font-weight: normal;
  color: #27566D;
  margin: 0;
  padding: 0;
}

<div
  class="c0"
  font-family="Gangster Grotesk, sans-serif"
  font-size="20"
  font-style="normal"
  font-weight="normal"
>
  title
</div>
`;
